src/components/ProseMirror.tsx

```diff
@@ -1,8 +1,25 @@
+import type { EditorState, Plugin, Transaction } from "prosemirror-state";
+import type { EditorProps, EditorView } from "prosemirror-view";
 import React from "react";
+import type { ReactNode } from "react";
 
+import { EditorProvider } from "../contexts/EditorContext.js";
 import { LayoutGroup } from "../contexts/LayoutGroup.js";
+import { useEditorView } from "../hooks/useEditorView.js";
 
-import { ProseMirrorInner, ProseMirrorProps } from "./ProseMirrorInner.js";
+interface Props extends EditorProps {
+  mount: HTMLElement | null;
+  children?: ReactNode | null;
+  defaultState?: EditorState;
+  state?: EditorState;
+  plugins?: readonly Plugin[];
+  dispatchTransaction?(this: EditorView, tr: Transaction): void;
+}
```

Comment:
This is repeated, rather than imported. I'd like to leave it that way for the moment because I think maybe we should consider exporting only the `<ProseMirror />` component, and hiding the hook, like we do for `<LayoutGroup />`.

Reply:
I think I agree about hiding the hook. I think we should still be able to import the props from the hook, though? I guess I don't see why hiding the hook would entail not being able to share the props, but I suspect I'm missing something.

Reply:
I think what I'm seeing is possibly just moving it all to one file next.